tensorflow/lite/kernels/internal/reference/sub.h

```diff
@@ -376,19 +265,37 @@ inline void BroadcastSub16POTSlow(const ArithmeticParams& params,
   NDOpsHelper<N>(output_desc, sub_func);
 }
 
-// Element-wise Sub that can often be used for inner loop of broadcast sub as
-// well as the non-broadcast sub.
-inline void SubElementwise(int size, const ArithmeticParams& params,
-                           const uint8_t* input1_data,
-                           const uint8_t* input2_data, uint8_t* output_data) {
-  TFLITE_DCHECK_GT(params.input1_offset, -256);
-  TFLITE_DCHECK_GT(params.input2_offset, -256);
-  TFLITE_DCHECK_LT(params.input1_offset, 256);
-  TFLITE_DCHECK_LT(params.input2_offset, 256);
+template <typename T, int N = 5>
+void BroadcastQuantSubSlow(const ArithmeticParams& params,
```

Comment:
can we split this into other changes? thanks!

Reply:
We could eventually separate the commits in the PR into multiple PRs as only the first commit fb9031aaac099feb3d58e5a59e2003890eef5982 changes the rounding. But the rounding change affects the tests and the DIV/SUB kernels. If we separate them in multiple PRs the CI will fail. I think it'd be easier to review each commit change one by one and still keep one PR.

Reply:
I think we should do this incrementally: 1 pr to fix the sub/div tests scaling issue and another to update MultiplyByQuantizedMuliplier and adjust for rounding issue